website/docs/opengl/c1/3-hello-texture.md

```diff
@@ -0,0 +1,447 @@
+
+If you do everything rightyou will see this result!
+
+![Quad with UVs](../../../images/opengl/chapter1/lesson3/quad-with-uvs.png)
+
+But we aren't there just yet. We can't see any texture in our quad!
+What you are seeing is the texture coordinates we passed displayed as a color!
+
+The Red, Green and mix between these two colors you see are, respectively, the X and Y texture coordinate values of that pixel.
+In the shader, the values are being read like this:
+
+![UV values](../../../images/opengl/chapter1/lesson3/quad-with-uvs-and-numbers.png)
+
+As you can see, the higher the U coordinate, the redder the pixel, and the higher the V coordinate, the greener the pixel.
+Even though we only specified UV values for each vertex, all pixels in the quad have UV values! That's because, as you read before,
+the fragment shader interpolates the coordinates for us, saving us a lot of work on our end.
+If we follow the diagonal from (0, 0) to (1, 1), we get the red and green colors growing together. The result of adding together
+red and green in the RGB color space is yellow!
+
+## Importing images as textures
+Now for the fun part: rendering an image!
+
+To start this section, first download
+[silk.png](https://github.com/dotnet/Silk.NET/blob/477154b99564aac9cf82426754857103fb88d2ea/examples/CSharp/OpenGL%20Tutorials/Tutorial%201.3%20-%20Textures/silk.png).
+We will use this image in this tutorial!
+
+In most cases with OpenGL, textures are uploaded as a sequence of bytes, this is usually done with a layout of 4 bytes per pixel (Red, Green, Blue, and Alpha).
+The hard part is: we can't just upload the bytes of a `.png` or `.jpg` file! These formats have a lot of unnecessary data, like headers and most importantly
+compression, which the GPU generally cannot understand.
+
+To load an image file as a byte array, we will first need a external library. In our case, we will use StbImageSharp.
+
+First of all, install the library using NuGet. To do so, you can use your IDE's NuGet package manager,
+or you can run the following command inside your project folder:
+```
+dotnet add package StbImageSharp
+```
+
+After installing, add the following reference on the top of your code file:
+```c#
+using StbImageSharp;
+```
+
+At the top of your class, declare a `uint` to store the ID of the OpenGL texture object:
+```c#
+private static uint _texture;
+```
+
+Returning to your `OnLoad` method, add this line at the end of the method to create, and bind the texture.
+That `ActiveTexture` call is telling OpenGL that we are wanting to use the first texture unit.
+```c#
+_texture = _gl.GenTexture();
+_gl.ActiveTexture(TextureUnit.Texture0);
+_gl.BindTexture(TextureTarget.Texture2D, _texture);
+```
+<?# info "A texture unit is a space in memory that refers to our texture object. Instead bind the texture object to the sampler, we bind the texture unit!" /?>
```

Comment:
```suggestion
<?# info "Texture units are locations in the OpenGL state where textures can be bound. Instead of textures being specified directly to shader samplers, textures are bound to a texture unit, and the texture unit is then specified to the shader sampler." /?>
```